networkx/algorithms/euler.py

```diff
@@ -246,16 +246,26 @@ def has_eulerian_path(G):
     eulerian_path
     """
     if G.is_directed():
+        nodes_remove = [v for v in G if G.in_degree[v] == 0 and G.out_degree[v] == 0]
```

Comment:
A comment here on why the if-statement might be worthwhile for future readers of the code, something like `# Remove isolated nodes (if any) without altering input graph`